### ISSUE TITLE
The time zone with the identifier 'UTC' was not found

### ISSUE THREAD

**Issue by Dim0N22** [Octonica/ClickHouseClient#40]
Hi and thanks for the library!

The library fails for me on `Open` when it tries to get server time zone, which is `UTC` in my configuration.
The exception is:
```
System.TimeZoneNotFoundException: The time zone with the identifier 'UTC' was not found.
   at Octonica.ClickHouseClient.Utils.TimeZoneHelper.GetTimeZoneInfo(String timeZone)
   at Octonica.ClickHouseClient.Types.DateTimeTypeInfo.Configure(ClickHouseServerInfo serverInfo)
```

I checked the [failing lines](https://github.com/Octonica/ClickHouseClient/blob/master/src/Octonica.ClickHouseClient/Utils/TimeZoneHelper.Net6.0.cs#L35):
```
if (!TimeZoneInfo.TryConvertIanaIdToWindowsId(timeZone, out windowsId))
   throw new TimeZoneNotFoundException("The time zone with the identifier '" + timeZone + "' was not found.");
```
and indeed `TryConvertIanaIdToWindowsId` doesn't find `UTC` on my Windows 10 system (running on .Net 6). 

The 
```
timeZoneInfo = TimeZoneInfo.FindSystemTimeZoneById(windowsId)
```
works though, so I guess instead of throwing an exception we could fallback to `FindSystemTimeZoneById` and fail if it can't find timezone too?

**Comment by SergeyMirvoda** [comment]
>Hi and thanks for the library!

Hi, you are welcome!

AFAIK we use the special library to handle timezone names for us.

**Comment by victor-sushko** [comment]
> AFAIK we use the special library to handle timezone names for us? Something changed with .net 6?

There is the method `TryConvertIanaIdToWindowsId` from the type `System.TimeZoneInfo` introduced in .NET 6. This method handles timezone conversions. So ClickHouseClient for .NET 6 no more needs the dependency of [TimeZoneConverter](https://www.nuget.org/packages/TimeZoneConverter/).

> and indeed `TryConvertIanaIdToWindowsId` doesn't find `UTC` on my Windows 10 system (running on .Net 6)

I checked the behavior of this method on several machines. Here is the code:
```C#
TimeZoneInfo.TryConvertIanaIdToWindowsId("UTC", out var windowsTzId);
Console.WriteLine($"Windows TZ for 'UTC': {windowsTzId}");

TimeZoneInfo.TryConvertIanaIdToWindowsId("Etc/UTC", out windowsTzId);
Console.WriteLine($"Windows TZ for 'Etc/UTC': {windowsTzId}");
```

The output was the same on every machine:
```
Windows TZ for 'UTC': UTC
Windows TZ for 'Etc/UTC': UTC
```

The link to the playground: https://dotnetfiddle.net/guAcv9

@Dim0N22, could you please run this code on your machine and share the result?

**Comment by Dim0N22** [comment]
Ok, I think I figured it out.

Dotnet runtime always returns `null` when `GlobalizationMode.Invariant` is set to true:
https://github.com/dotnet/runtime/blob/3de620787ab3962dcb89664a691cf917653d5bd2/src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.FullGlobalizationData.cs#L10-L16

This is exactly my case, I have 
```
<PropertyGroup>
  <InvariantGlobalization>true</InvariantGlobalization>
</PropertyGroup>
```
in my `*.csproj` file. As far as I remember, I created it with a default .Net 6 template, so probably it'll affect more people in the future.

So while I can partly fix it by disabling `InvariantGlobalization`, I'm not sure it's the best solution for me as I need to run my code both on Windows and Linux and based on [the docs](https://github.com/dotnet/runtime/blob/main/docs/design/features/globalization-invariant-mode.md), Invariant Globalization makes it easier.

Having a fallback would still be useful to me.

**Comment by victor-sushko** [comment]
Thank you for your investigation!

It's really turned out that the entire OS check here was redundant!
https://github.com/Octonica/ClickHouseClient/blob/f7df0f2f956d85833e04f6be138760ffcef95b2f/src/Octonica.ClickHouseClient/Utils/TimeZoneHelper.Net6.0.cs#L32-L36

I also made a fix to mitigate similar errors. The `TimeZoneNotFoundException` is now thrown when a `DateTime(64)` column is met, not when the connection is opening.

The bug was fixed in the version [2.2.8](https://www.nuget.org/packages/Octonica.ClickHouseClient/2.2.8).
